### PR TITLE
Making go links public

### PR DESCRIPTION
**Pull request by saf6260**
Related to my PR on onerepo, this updates the backend to handle public go link creation

### CODE REVIEW

File: db/migrations/20150809163538-links.js [hunk 1]
```diff
@@ -9,6 +9,14 @@ export function up(queryInterface, Sequelize) {
       type: Sequelize.STRING,
       allowNull: false,
     },
+    goDescription: {
```
Comment (billdybas) [inline db/migrations/20150809163538-links.js]: You will want to make a new migration file that specifically adds these columns. Updating existing migration files is an anti-pattern.
Also, naming: `description` and `public`

File: db/migrations/20150809163538-links.js [hunk 2]
```diff
@@ -9,6 +9,14 @@ export function up(queryInterface, Sequelize) {
...
+      type: Sequelize.STRING,
+      allowNull: false,
+    },
+    publicGO: {
+      type: Sequelize.STRING,
```
Comment (billdybas) [inline db/migrations/20150809163538-links.js]: This should be a boolean.

File: db/migrations/20150809163538-links.js [hunk 3]
```diff
@@ -9,6 +9,14 @@ export function up(queryInterface, Sequelize) {
       type: Sequelize.STRING,
       allowNull: false,
     },
+    goDescription: {
+      type: Sequelize.STRING,
+      allowNull: false,
+    },
+    publicGO: {
```
Comment (billdybas) [inline db/migrations/20150809163538-links.js]: You should set a defaultValue of false.

File: db/migrations/20150809163538-links.js [hunk 4]
```diff
@@ -9,6 +9,14 @@ export function up(queryInterface, Sequelize) {
       type: Sequelize.STRING,
       allowNull: false,
     },
+    goDescription: {
+      type: Sequelize.STRING,
+      allowNull: false,
```
Comment (billdybas) [inline db/migrations/20150809163538-links.js]: This should be optional. Remove this.

File: models/link.js [hunk 5]
```diff
@@ -21,13 +21,29 @@ export default sequelize.define('links', {
...
+    },
+  },
 }, {
   scopes: {
     paginate,
     orderBy(field, direction) {
       return sorting(field, direction, [
         'shortLink',
         'longLink',
+        'goDescription',
```
Comment (billdybas) [inline models/link.js]: Ordering by description or public status doesn't seem useful. I'd remove this.

File: models/link.js [hunk 6]
```diff
@@ -21,13 +21,29 @@ export default sequelize.define('links', {
...
+  },
+  publicGO: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    validate: {
+      notEmpty: true
+    },
+  },
 }, {
   scopes: {
```
Comment (billdybas) [inline models/link.js]: Look at how we do scopes in other models. You should add one here for filtering links that are public.

File: package-lock.json [hunk 7]
```diff
@@ -1576,7 +1576,6 @@
       "requires": {
         "anymatch": "1.3.2",
         "async-each": "1.0.1",
-        "fsevents": "1.1.3",
```
Comment (billdybas) [inline package-lock.json]: I'm confused why all these lines got deleted. What version of node are you using?

**Comment by saf6260** [comment]
Apologies I meant to push this earlier. This should be updated now. I had to roll back to node 8.9.3 for sqlite to work properly, so thats probably why the package-lock got messed with. I will delete that file change before the merge that way the server isnt messed with

**Pull request merged**